scripts/Modules/mkw_utils.py

```diff
@@ -81,5 +122,215 @@ def get_unrounded_time(lap, player):
+    t2 = time_to_cross(P2, S2, P1, P1+S1)
+    return t1-t2
+
+
+def get_time_difference_crosspath(player1, player2):
+    """Time difference "CrossPath"
+    Take both XZ trajectories of the player and the ghost
+    Calculate how much time it takes them to reach the crosspoint. (2D only)
+    Return the difference."""
+    P1, S1, P2, S2 = get_physics(player1, player2)
+    return time_difference_crosspath(P1, P2, S1, S2)
+
+
+
+def get_finish_line_coordinate():
+    """pointA is the position of the left side of the finish line (vec3)
+        point B ---------------------right------------------------------
+        both have 0 as their Y coordinate."""
+    game_id = utils.get_game_id()
+    address = {"RMCE01": 0x809B8F28, "RMCP01": 0x809BD6E8,
+               "RMCJ01": 0x809BC748, "RMCK01": 0x809ABD28}
+    kmp_ref = chase_pointer(address[game_id], [0x4, 0x0], 'u32')
+    offset = memory.read_u32(kmp_ref+0x24)
+    pointA = vec3(memory.read_f32(kmp_ref+0x4C+offset+0x8+0x0), 0, memory.read_f32(kmp_ref+0x4C+offset+0x8+0x4))
+    pointB = vec3(memory.read_f32(kmp_ref+0x4C+offset+0x8+0x8), 0, memory.read_f32(kmp_ref+0x4C+offset+0x8+0xC))
+    return pointA, pointB
+
+
+def time_difference_tofinish(P1, P2, S1, S2):
+    A,B = get_finish_line_coordinate()
+    t1 = time_to_cross(P1, S1, A, B)
+    t2 = time_to_cross(P2, S2, A, B)
+    return t1-t2
+
+def get_time_difference_tofinish(player1, player2):
+    """Assume player and ghost are not accelerated.
+    Calculate the time to the finish line for both, and takes the difference."""
+    P1, S1, P2, S2 = get_physics(player1, player2)
+    return time_difference_tofinish(P1, P2, S1, S2)
+
+
+def find_index(value, value_list):
+    """Find the index i so value_list[i]>=value>value_list[i+1]
+        We suppose value_list[i+1] < value_list[i]
+            and value_list[0]>= value>=value_list[-1]"""
+    n = len(value_list)
+    if n == 1 :
+        return 0
+    h = n//2
+    if value <= value_list[h]:
+        return h+find_index(value, value_list[h:])
+    return find_index(value, value_list[:h])
+    
+def get_time_difference_racecompletion(history):
+    """Use RaceCompletionData History to calculate the frame difference
+        The function assume that RaceCompletion is increasing every frames"""
+    curframe = history.get_older_frame(0)
+    lastframe = history.get_older_frame(-1)
+    inf = float('inf')
+    if curframe.prc>=curframe.grc:
+        if curframe.grc>lastframe.prc:
+            l = [history.get_older_frame(k).prc for k in range(history.size)]
+            i = find_index(curframe.grc, l)
+            t = i + (curframe.grc - l[i])/ (l[i+1] - l[i])
+            return -t
+        return -inf
+    else:
+        if curframe.prc>lastframe.grc:
+            l =[history.get_older_frame(k).grc for k in range(history.size)]
+            i = find_index(curframe.prc, l)
+            t = i + (curframe.prc - l[i])/ (l[i+1] - l[i])
+            return t
+        return inf
```

Comment:
I believe the linter does not like some of the variable names, however this should fix it.
```suggestion
    if curframe.prc>=curframe.grc:
        if curframe.grc>lastframe.prc:
            length = [history.get_older_frame(k).prc for k in range(history.size)]
            i = find_index(curframe.grc, l)
            time = i + (curframe.grc - length[i])/ (length[i+1] - length[i])
            return -time
        return -inf
    else:
        if curframe.prc>lastframe.grc:
            length = [history.get_older_frame(k).grc for k in range(history.size)]
            i = find_index(curframe.prc, l)
            time = i + (curframe.prc - length[i])/ (length[i+1] - length[i])
            return time
        return inf
```